scripts/check-deployed-dssspell.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Define colors
+GREEN='\033[1;32m'
+RED='\033[1;31m'
+NC='\033[0m' # No Color
+
+function success_check() {
+  echo -e "[${GREEN}✔${NC}] ${GREEN}$1${NC}"
+}
+
+function error_check() {
+  echo -e "[${RED}✖${NC}] ${RED}$1${NC}"
+}
+
+[[ "$(cast chain --rpc-url="$ETH_RPC_URL")" == "goerli" ]] || { echo -e "Please set a Goerli ETH_RPC_URL"; exit 1; }
+[[ "$ETHERSCAN_API_KEY" ]] || { echo -e "Please set ETHERSCAN_API_KEY"; exit 1; }
+
+# Etherscan API endpoint
+ETHERSCAN_API="https://api-goerli.etherscan.io/api"
+
+# Path to config.sol file
+CONFIG_PATH="src/test/config.sol"
+
+# DssSpell data
+LICENSE="AGPL-3.0-or-later"
+SOLC="v0.8.16+commit.07a7930e"
+
+# Read contract address, block number, and timestamp from Solidity source code
+deployed_spell_address=$(grep -oE 'deployed_spell:\s+address\((0x[a-fA-F0-9]+)\)' $CONFIG_PATH | sed -E 's/^.*\((.*)\)/\1/')
+deployed_spell_block=$(grep -oE 'deployed_spell_block\s*:\s*[0-9]+' $CONFIG_PATH | sed -E 's/.*:\s*//')
+deployed_spell_timestamp=$(grep -oE 'deployed_spell_created\s*:\s*[0-9]+' $CONFIG_PATH | sed -E 's/.*:\s*//')
+
+# Check if contract address, block number, and timestamp are zero
+if [[ "$deployed_spell_address" =~ ^(address\(0\)|0)$ ]] || [[ "$deployed_spell_block" = "0" ]] || [[ "$deployed_spell_timestamp" = "0" ]]; then
+  echo "DssSpell address, block number, or timestamp is not set in config file."
+  exit 1
+fi
+
+# Get contract verification information
+verified_spell_info=$(curl -s "$ETHERSCAN_API?module=contract&action=getsourcecode&address=$deployed_spell_address" | jq -r .result[0])
+
+# Check contract verification status
+verified=$(echo "$verified_spell_info" | jq -r '.SourceCode != null')
+if [ "$verified" ]; then
+  success_check "DssSpell is verified."
+else
+  error_check "DssSpell not verified."
+fi
+
+# # Check license type
+# license=$(echo "$verified_spell_info" | jq -r '.LicenseType')
+# if [ "$license" == "$LICENSE" ]; then
+#   success_check "DssSpell was verified with a valid license."
+# else
+#   error_check "DssSpell was verified with an invalid or unknown license."
+# fi
```

Comment:
is this not needed anymore?

Reply:
It should be fixed now, uncommented in 5ead6dd